### PR TITLE
Add Elastikube - a EKS-like Kubernetes service built with pure terraform

### PR DESCRIPTION
**Pull request by alanchchen**

### CODE REVIEW

File: aws/kube-etcd/role.tf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action" : [
+        "s3:GetObject"
+      ],
+      "Resource": "arn:aws:s3:::*",
```
Comment (smalltown) [inline aws/kube-etcd/role.tf]: Limit the get permission for only the specific s3 bucket?

File: aws/kube-master/lb.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    self        = true
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (smalltown) [inline aws/kube-master/lb.tf]: How about only the master vpc cidr can access the lb?

File: aws/kube-master/worker-sg.tf [hunk 3]
```diff
@@ -0,0 +1,50 @@
+resource "aws_security_group" "workers" {
```
Comment (smalltown) [inline aws/kube-master/worker-sg.tf]: why worker sg put in the mater folder?

File: aws/elastikube/variables.tf [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    (Optional) List of security group IDs for the cross-account elastic network interfaces
+    to use to allow communication between your worker nodes and the Kubernetes control plane.
+EOF
+}
+
+variable "subnet_ids" {
+  type    = "list"
+  default = []
+
+  description = <<EOF
```
[the remark this answers is not in the document]
Comment (alanchchen) [reply aws/elastikube/variables.tf]: For the security concern, yes.

File: aws/kube-etcd/role.tf (hunk 1 above)
Comment (alanchchen) [reply aws/kube-etcd/role.tf]: ok

File: aws/kube-master/lb.tf (hunk 2 above)
Comment (alanchchen) [reply aws/kube-master/lb.tf]: ok

**Pull request merged**